### PR TITLE
implemented an async ingester for aggregates to avoid http timeouts

### PR DESCRIPTION
**Pull request by SchawnnDev**

### CODE REVIEW

File: config/engine-api.toml [hunk 1]
```diff
@@ -116,6 +116,10 @@ POSTGRESQL_CONN_MAX_LIFETIME = "0"
 # Default value: "true"
 ENABLE_CRONS_ON_START = "true"
 
+# Specify the aggregate-ingesters maximum queue size
+# Default value: "100"
+AGGREGATEINGESTER_QUEUE_BUFFER_SIZE = "100"
```
Comment (SchawnnDev) [inline config/engine-api.toml:121]: Is 100 enough or too much?

**Comment by SchawnnDev** [comment]
Tested in https://github.com/myrteametrics/myrtea-connector-api-colissimo-qssla/pull/26
Working fine

File: internals/ingester/aggregate.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	for ir := range ingester.Data {
+		zap.L().Debug("Received ExternalAggregate", zap.Int("ExternalAggregate items count", len(ir)))
+
+		err := HandleAggregates(ir)
+		if err != nil {
+			zap.L().Error("AggregateIngester: could not handle aggregates", zap.Error(err))
+		}
+
+		// Update queue gauge
+		(*ingester.metricQueueGauge).Set(float64(len(ingester.Data)))
```
Comment (cedricablain) [inline internals/ingester/aggregate.go:60]: Possible de faire directement ça ?
```suggestion
		ingester.metricQueueGauge.Set(float64(len(ingester.Data)))
```

**Review by cedricablain — APPROVED** [review]

File: config/engine-api.toml (hunk 1 above)
Comment (cedricablain) [reply config/engine-api.toml:121]: Ok for 100 by default

**Comment by cedricablain** [comment]
> Question: only one ingester or multiple for each route endpoint created ? Actually two instances are created in :
> 
> ```
> 	case "BASIC":
> 		routes, err = buildRoutesV3Basic(config)
> 	case "SAML":
> 		routes, err = buildRoutesV3SAML(config)
> ```

@SchawnnDev Only one ingester is instantiated ; either in BASIC auth mode, or in SAML auth mode.

**Pull request merged**